### PR TITLE
feat: add --ssl-insecure-skip-verify flag which we can set in test sc…

### PR DESCRIPTION
**Pull request by rawlingsj**
…enarious and prevents sso proxy pods failing to start

### CODE REVIEW

**Comment by rawlingsj** [comment]
Default behaviour is still the recommended secure way but this gives us the option in CI scenarios to skip the check and verify the environment.

File: pkg/apis/jenkins.io/v1/types.go [hunk 1]
```diff
@@ -46,6 +46,8 @@ type SSOSpec struct {
 	URLTemplate string `json:"urlTemplate,omitempty"`
 	// SkipExposeService to avoid using exposecontroller to create ingress rule for proxy
 	SkipExposeService bool `json:"skipExposeService,omitempty"`
+	// SSLInsecureSkipVerify allows the proxy container to connect with a OIDC using selft-sogned certs, this should be used for testing only
+	SSLInsecureSkipVerify bool `json:"sslInsecureSkipVerify,omitempty"`
```
Comment (pmuir) [inline pkg/apis/jenkins.io/v1/types.go]: Shouldn't this be `tls`?

Comment (rawlingsj) [reply pkg/apis/jenkins.io/v1/types.go]: It maps to the same flag that's set on for the proxy https://github.com/rawlingsj/sso-operator/blob/c561ba6b9b3caeaf7f9c19bd9fe90f24003d85dc/pkg/proxy/proxy.go#L325 so maybe `ProxySSLInsecureSkipVerify`?

Comment (pmuir) [reply pkg/apis/jenkins.io/v1/types.go]: Fair enough :-)

**Pull request merged**